//
// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package scql.engine.services.pb;

option cc_generic_services = true;

import "api/status.proto";

service ErrorCollectorService {
  // report error to peer engines
  rpc ReportError(ReportErrorRequest) returns (ReportErrorResponse);
}

message ReportErrorRequest {
  string session_id = 1;
  string party_code = 2;
  scql.pb.Status status = 3;
}

message ReportErrorResponse {
  scql.pb.Status status = 1;
}
